extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -360,6 +379,51 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
     });
   }
 
+  /**
+   * Handles Fluid-related messages dispatched from SafeFrame.
+   * @param {!Event} event
+   */
+  receiveMessageForFluid(event) {
+    if (event.origin != 'https://tpc.googlesyndication.com') {
+      return;
+    }
+    const data = tryParseJson(getData(event));
+    if (!data || data['s'] != 'creative_geometry_update' ||
+        data['sentinel'] != this.sentinel) {
+      return;
+    }
+    // The first creative_geometry_update message will contain bad
+    // geometric data, as it will have been computed using the initial,
+    // incorrect, iframe style. We use this first message as a
+    // triggering point to restyle the iframe, which will in turn cause
+    // safeframe to send another creative_geometry_update message, this
+    // time containing the correct information.
+    const styleString = this.iframe.getAttribute('style');
+    if (/width: 0px/.test(styleString) &&
+        /height: 0px/.test(styleString)) {
+      setStyles(this.iframe, {
+        width: '100%',
+        height: '100%',
+        position: 'relative',
+      });
+      setStyles(this.element, {
+        width: computedStyle(this.win,
```

Comment:
You're causing a complete style recalc here!

Reply:
Not sure I have any other option. Please see the days-long back-and-forth I had with Dima about this.

Reply:
Why do we need to set the width at all? If it's already `position: absolute` with `display: block`, we have a responsive width. No need for any recalcs.

Reply:
Without a width on the ad slot, safeframe is unable to measure its geometry correctly, or more specifically its height (since the width is defined).